### PR TITLE
Sending start services log after enable

### PR DESCRIPTION
**Pull request by MatkovIvan**

### CODE REVIEW

File: AppCenter/AppCenter/MSAppCenter.m [hunk 1]
```diff
@@ -16,13 +16,11 @@
 #import "MSCustomPropertiesPrivate.h"
 #endif
 
-// Singleton
+// Singleton.
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.m]: This is a comment for **definition** of variables. Should use `/** */`

File: AppCenter/AppCenter/MSAppCenter.m [hunk 2]
```diff
@@ -16,13 +16,11 @@
...
 static MSAppCenter *sharedInstance = nil;
 static dispatch_once_t onceToken;
 
-/**
- * Base URL for HTTP Ingestion backend API calls.
- */
+// Base URL for HTTP Ingestion backend API calls.
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.m]: Revert the change due to above comment.

File: AppCenter/AppCenter/MSAppCenter.m (hunk 1 above)
Comment (jaelim-ms) [reply AppCenter/AppCenter/MSAppCenter.m]: Please change the other two below. The scope for the change is remaining only on this file. We can do others with any changes of files.

File: AppCenter/AppCenter/MSAppCenter.m [hunk 3]
```diff
@@ -313,11 +311,11 @@ - (void)setEnabled:(BOOL)isEnabled {
   self.enabledStateUpdating = YES;
   if ([self isEnabled] != isEnabled) {
 
-    // Enable/disable pipeline.
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.m]: Any intentions of this change?

File: AppCenter/AppCenter/MSAppCenter.m [hunk 4]
```diff
@@ -418,15 +422,18 @@ - (void)sendStartServiceLog:(NSArray<NSString *> *)servicesNames {
...
+    }
+    [self.startedServicesNamesToLog addObjectsFromArray:servicesNames];
   }
 }
 
 #if !TARGET_OS_TV
 - (void)sendCustomPropertiesLog:(NSDictionary<NSString *, NSObject *> *)properties {
   MSCustomPropertiesLog *customPropertiesLog = [MSCustomPropertiesLog new];
   customPropertiesLog.properties = properties;
-
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.m]: Thanks for the cleanup!

File: AppCenter/AppCenter/MSAppCenter.m [hunk 5]
```diff
@@ -418,15 +422,18 @@ - (void)sendStartServiceLog:(NSArray<NSString *> *)servicesNames {
     MSStartServiceLog *serviceLog = [MSStartServiceLog new];
     serviceLog.services = servicesNames;
     [self.logManager processLog:serviceLog forGroupId:kMSGroupId];
+  } else {
+    if (self.startedServicesNamesToLog == nil) {
+      self.startedServicesNamesToLog = [NSMutableArray new];
+    }
+    [self.startedServicesNamesToLog addObjectsFromArray:servicesNames];
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/MSAppCenter.m]: Can we just point servicesNames instead of creating an array?

File: AppCenter/AppCenter/Internals/MSAppCenterInternal.h [hunk 6]
```diff
@@ -19,6 +19,7 @@ static NSString *const kMSDisableAll = @"All";
 
 @property(nonatomic) id<MSLogManager> logManager;
 @property(nonatomic) NSMutableArray<NSObject<MSServiceInternal> *> *services;
+@property(nonatomic) NSMutableArray<NSString *> * startedServicesNamesToLog;
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/MSAppCenterInternal.h]: Just `startedServiceNames`?

File: AppCenter/AppCenter/MSAppCenter.m (hunk 5 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/MSAppCenter.m]: Nope. I've added a test for this case. See https://github.com/Microsoft/AppCenter-SDK-Apple/pull/886/commits/1986a03397e878983645079b4d9f22e8d27b775f

File: AppCenter/AppCenter/MSAppCenter.m (hunk 3 above)
Comment (MatkovIvan) [reply AppCenter/AppCenter/MSAppCenter.m]: Because `isEnabled` is checked in `applyPipelineEnabledState` (`sendStartServiceLog`)

Comment (jaelim-ms) [reply AppCenter/AppCenter/MSAppCenter.m]: The `isEnabled` is a parameter passed to `applyPipelineEnabledState`. It doesn't read anything in NSUserDefaults. I think we should make sure the pipeline is properly changed before setting the value to NSUserDefaults.

Comment (jaelim-ms) [reply AppCenter/AppCenter/MSAppCenter.m]: Changing this is not a big deal but I would like to keep the previous code to get rid of a chance for side effect. Please revert unless it has to be changed.

Comment (jaelim-ms) [reply AppCenter/AppCenter/MSAppCenter.m]: OK I got it. `isEnabled` from `sendStartServiceLog`.
